src/analysis/normed_space/exponential.lean

```diff
@@ -0,0 +1,551 @@
+/-
+Copyright (c) 2021 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+import analysis.calculus.deriv
+import analysis.specific_limits
+import data.complex.exponential
+import analysis.complex.basic
+import topology.metric_space.cau_seq_filter
+
+/-!
+# Exponential in a Banach algebra
+
+In this file, we define `exp 𝕂 𝔸`, the exponential map in a normed algebra `𝔸` over a nondiscrete
+normed field `𝕂`. Although the definition doesn't require `𝔸` to be complete, we need to assume it
+for most results.
+
+We then prove basic results, as described below.
+
+## Main result
+
+We prove most result for an arbitrary field `𝕂`, and then specialize to `𝕂 = ℝ` or `𝕂 = ℂ`.
+
+### General case
+
+- `has_strict_fderiv_at_exp_zero_of_radius_pos` : `exp 𝕂 𝔸` has strict Fréchet-derivative
+  `1 : 𝔸 →L[𝕂] 𝔸` at zero, as long as it converges on a neighborhood of zero
+  (see also `has_strict_deriv_at_exp_zero_of_radius_pos` for the case `𝔸 = 𝕂`)
+- `exp_add_of_commute_of_lt_radius` : if `𝕂` has characteristic zero, then given two commuting
+  elements `x` and `y` in the disk of convergence, we have
+  `exp 𝕂 𝔸 (x+y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y)`
+- `exp_add_of_lt_radius` : if `𝕂` has characteristic zero and `𝔸` is commutative, then given two
+  elements `x` and `y` in the disk of convergence, we have
+  `exp 𝕂 𝔸 (x+y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y)`
+- `has_strict_fderiv_at_exp_of_lt_radius` : if `𝕂` has characteristic zero and `𝔸` is commutative,
+  then given a point `x` in the disk of convergence, `exp 𝕂 𝔸` as strict Fréchet-derivative
+  `exp 𝕂 𝔸 x • 1 : 𝔸 →L[𝕂] 𝔸` at x (see also `has_strict_deriv_at_exp_of_lt_radius` for the case
+  `𝔸 = 𝕂`)
+
+### `𝕂 = ℝ` or `𝕂 = ℂ`
+
+- `exp_series_radius_eq_top` : the `formal_multilinear_series` defining `exp 𝕂 𝔸` has infinite
+  radius of convergence
+- `has_strict_fderiv_at_exp_zero` : `exp 𝕂 𝔸` has strict Fréchet-derivative `1 : 𝔸 →L[𝕂] 𝔸` at zero
+  (see also `has_strict_deriv_at_exp_zero` for the case `𝔸 = 𝕂`)
+- `exp_add_of_commute` : given two commuting elements `x` and `y`, we have
+  `exp 𝕂 𝔸 (x+y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y)`
+- `exp_add` : if `𝔸` is commutative, then we have `exp 𝕂 𝔸 (x+y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y)`
+  for any `x` and `y`
+- `has_strict_fderiv_at_exp` : if `𝔸` is commutative, then given any point `x`, `exp 𝕂 𝔸` as strict
+  Fréchet-derivative `exp 𝕂 𝔸 x • 1 : 𝔸 →L[𝕂] 𝔸` at x (see also `has_strict_deriv_at_exp` for the
+  case `𝔸 = 𝕂`)
+
+### Other useful compatibility results
+
+- `exp_eq_exp_of_field_extension` : given `𝕂' / 𝕂` a normed field extension (that is, an instance
+  of `normed_algebra 𝕂 𝕂'`) and a normed algebra `𝔸` over both `𝕂` and `𝕂'`, if `𝕂` and `𝕂'` have
+  the same characteristic and have compatible actions on `𝔸`, then `exp 𝕂 𝔸 = exp 𝕂' 𝔸`
```

Comment:
```suggestion
  of `normed_algebra 𝕂 𝕂'`) and a normed algebra `𝔸` over both `𝕂` and `𝕂'` then 
  `exp 𝕂 𝔸 = exp 𝕂' 𝔸`
```
Your characteristic assumption is automatic.

Reply:
Indeed, thanks !